dtests/src/test/java/io/snappydata/hydra/security/SnappySecurityTest.java

```diff
@@ -0,0 +1,464 @@
+                    execute(queryStr, conn);
+                  }
+                }
+                else {
+                  if (schemaToTest.containsAll(patternMatch)) {
+                    isAuth = true;
+                    Log.getLogWriter().info("The user " + usr + "will execute the Join/Sub query   " + queryStr + " with new authorization = " + isAuth);
+                    execute(queryStr, conn);
+                  } else {
+                    isAuth = false;
+                    Log.getLogWriter().info("The user " + usr + "will execute the Join/Sub query   " + queryStr + " with new authorization = " + isAuth);
+                    execute(queryStr, conn);
+                  }
+                }
+              }
+            }
+          } else {
+            for (int s = 0; s < schemaToTest.size(); s++) {
+              String str = schemaToTest.elementAt(s).toString();
+              Log.getLogWriter().info("Find " + str + " in query " + queryStr);
+              if (!queryStr.contains(str)) {
+                isAuth = false;
+                Log.getLogWriter().info("The user " + usr + "will execute the query   " + queryStr + " with new authorization = " + isAuth);
+                execute(queryStr, conn);
+              } else {
+                if (usr.equals(unAuthUser))
+                  isAuth = false;
+                else
+                  isAuth = true;
+                Log.getLogWriter().info("The user " + usr + "will execute the query   " + queryStr + " with new authorization = " + isAuth);
+                execute(queryStr, conn);
+              }
+            }
+          }
+        } else {
+          Log.getLogWriter().info("The query to be executed is  " + queryStr + " with new authorization = " + isAuth);
+          execute(queryStr, conn);
+        }
+      } catch (SQLException e) {
+        if (isAuth) {
+          unExpectedExceptionCnt = unExpectedExceptionCnt + 1;
+          Log.getLogWriter().info(" unExpectedExceptionCnt Count is " + unExpectedExceptionCnt);
+          Log.getLogWriter().info("Got UnExpected Exception " + e.getMessage());
+        } else {
+          expectedExceptionCnt = expectedExceptionCnt + 1;
+          Log.getLogWriter().info("Got Expected Exception " + e.getMessage());
+        }
+      }
+    }
+    closeConnection(conn);
+  }
+
+  public static void execute(String queryStr, Connection conn) throws SQLException {
+    if (queryStr.contains("SELECT"))
+      conn.createStatement().executeQuery(queryStr);
+    else
+      conn.createStatement().execute(queryStr);
+    Log.getLogWriter().info("Query executed successfully");
+  }
+
+  public static void validate(Integer expectedCnt, Integer unExpectedCnt) {
+    if (unExpectedExceptionCnt != unExpectedCnt)
+      throw new TestException("The Result is WRONG :Expected unExpectedExceptionCnt = " + unExpectedCnt + " but got " +
```

Comment:
Confusing parameter name and log statement. Expected unExpectedExceptionCount. It should be Expected exception count.